### ISSUE TITLE
test-random.R Has Multiple Issues on Windows

### ISSUE THREAD

**Issue by billdenney** [nlmixr2/rxode2#26]
The log of output is below (I plan to take a shot at fixing it, but it is going to take different/more work than the other simpler fixes in #19.

```
==> Testing R file using 'testthat'

i Loading rxode2
rxode2 1.0.0 using 4 threads (see ?getRxThreads)
  no cache: create with `rxCreateCache()`

== Testing test-random.R =======================================================
[ FAIL 0 | WARN 0 | SKIP 0 | PASS 0 ][====|====|===
=|====|====|====|====|====|====|====] 0:00:00 
[ FAIL 6 | WARN 3 | SKIP 0 | PASS 1 ][====|====|====|====|====|====|===
=|====|====|====] 0:00:00 
[ FAIL 6 | WARN 3 | SKIP 0 | PASS 8 ]rxode2 model syntax error:
================================================================================
:001: 'rinorm'/'rxnorm'/'rnorm' takes 0-2 arguments 'rxnorm(mean, sd)':
          x4 <- rnorm(a, b, c, d)
                ^~~~~
================================================================================
[ FAIL 6 | WARN 3 | SKIP 0 | PASS 13 ]rxode2 model syntax error:
================================================================================
:001: 'rinormV'/'rxnormV'/'rnormV' takes 0-2 arguments 'rxnormV(mean, sd)':
          x4 <- rnormV(a, b, c, d)
                ^~~~~~
================================================================================
[ FAIL 9 | WARN 5 | SKIP 0 | PASS 21 ]rxode2 model syntax error:
================================================================================
:001: 'ricauchy'/'rxcauchy'/'rcauchy' takes 0-2 arguments 'rxcauchy(location, scale)':
          x4 <- rcauchy(a, b, c, d)
                ^~~~~~~
================================================================================
[ FAIL 13 | WARN 7 | SKIP 0 | PASS 25 ]rxode2 model syntax error:
================================================================================
:001: 'rchisq' takes 1 argument:
          x1 <- rchisq()
                       ^
================================================================================
rxode2 model syntax error:
================================================================================
:001: 'rchisq' takes 1 argument:
          x1 <- rchisq(a, b)
                ^~~~~~
================================================================================
[ FAIL 17 | WARN 9 | SKIP 0 | PASS 30 ]rxode2 model syntax error:
================================================================================
:001: 'rexp' takes 0-1 argument:
          x1 <- rexp(a, b)
                ^~~~
================================================================================
[ FAIL 21 | WARN 11 | SKIP 0 | PASS 34 ]rxode2 model syntax error:
================================================================================
:001: 'rif'/'rxf'/'rf' takes 2 arguments 'rxf(df1, df2)':
          x1 <- rf(a, b, c)
                ^~
================================================================================
[ FAIL 21 | WARN 11 | SKIP 0 | PASS 35 ]rxode2 model syntax error:
================================================================================
:001: 'rif'/'rxf'/'rf' takes 2 arguments 'rxf(df1, df2)':
          x1 <- rf(a)
                   ^
================================================================================
[ FAIL 21 | WARN 11 | SKIP 0 | PASS 36 ]rxode2 model syntax error:
================================================================================
:001: 'rif'/'rxf'/'rf' takes 2 arguments 'rxf(df1, df2)':
          x1 <- rf()
                   ^
================================================================================
[ FAIL 23 | WARN 11 | SKIP 0 | PASS 40 ]rxode2 model syntax error:
================================================================================
:001: 'rigamma'/'rxgamma'/'rgamma' takes 1-2 arguments 'rxgamma(shape, rate)':
          x1 <- rgamma(a, b, c)
                ^~~~~~
================================================================================
[ FAIL 23 | WARN 11 | SKIP 0 | PASS 41 ]rxode2 model syntax error:
================================================================================
:001: 'rigamma'/'rxgamma'/'rgamma' takes 1-2 arguments 'rxgamma(shape, rate)':
          x1 <- rgamma()
                       ^
================================================================================
[ FAIL 27 | WARN 13 | SKIP 0 | PASS 45 ]rxode2 model syntax error:
================================================================================
:001: 'ribeta'/'rxbeta'/'rbeta' takes 2 arguments 'rxbeta(shape1, shape2)':
          x1 <- rbeta(a, b, c)
                ^~~~~
================================================================================
[ FAIL 27 | WARN 13 | SKIP 0 | PASS 46 ]rxode2 model syntax error:
================================================================================
:001: 'ribeta'/'rxbeta'/'rbeta' takes 2 arguments 'rxbeta(shape1, shape2)':
          x1 <- rbeta(a)
                      ^
================================================================================
[ FAIL 27 | WARN 13 | SKIP 0 | PASS 47 ]rxode2 model syntax error:
================================================================================
:001: 'ribeta'/'rxbeta'/'rbeta' takes 2 arguments 'rxbeta(shape1, shape2)':
          x1 <- rbeta()
                      ^
================================================================================
[ FAIL 28 | WARN 13 | SKIP 0 | PASS 50 ]rxode2 model syntax error:
================================================================================
:001: 'rigeom'/'rxgeom'/'rgeom' takes 1 argument 'rxgeom(prob)':
          x1 <- rgeom()
                      ^
================================================================================
[ FAIL 28 | WARN 13 | SKIP 0 | PASS 52 ]rxode2 model syntax error:
================================================================================
:001: 'rigeom'/'rxgeom'/'rgeom' takes 1 argument 'rxgeom(prob)':
          x1 <- rgeom(a, b)
                ^~~~~
================================================================================
[ FAIL 32 | WARN 15 | SKIP 0 | PASS 56 ]rxode2 model syntax error:
================================================================================
:001: 'ripois'/'rxpois'/'rpois' takes 1 argument 'rxpois(lambda)':
          x1 <- rpois()
                      ^
================================================================================
[ FAIL 32 | WARN 15 | SKIP 0 | PASS 57 ]rxode2 model syntax error:
================================================================================
:001: 'ripois'/'rxpois'/'rpois' takes 1 argument 'rxpois(lambda)':
          x1 <- rxpois(a, b)
                ^~~~~~
================================================================================
[ FAIL 36 | WARN 17 | SKIP 0 | PASS 61 ]rxode2 model syntax error:
================================================================================
:001: 'rt' takes 1 argument:
          x1 <- rt()
                   ^
================================================================================
[ FAIL 36 | WARN 17 | SKIP 0 | PASS 62 ]rxode2 model syntax error:
================================================================================
:001: 'rt' takes 1 argument:
          x1 <- rt(a, b)
                ^~
================================================================================
[====|====|====|==
[ FAIL 36 | WARN 18 | SKIP 0 | PASS 63 ]==|====|====|====|====|====|====] 0:00:00 
[ FAIL 42 | WARN 20 | SKIP 0 | PASS 63 ][====|====|====|====|====
[ FAIL 42 | WARN 20 | SKIP 0 | PASS 64 ]|====|====|====|====|====] 0:00:00 
[ FAIL 42 | WARN 20 | SKIP 0 | PASS 71 ]rxode2 model syntax error:
================================================================================
:001: 'riunif'/'rxunif'/'runif' takes 0-2 arguments 'rxunif(min, max)':
          x4 <- runif(a, b, c, d)
                ^~~~~
================================================================================
[ FAIL 46 | WARN 22 | SKIP 0 | PASS 76 ]rxode2 model syntax error:
================================================================================
:001: 'riweibull'/'rxweibull'/'rweibull' takes 1-2 arguments 'rxweibull(shape, scale)':
          x1 <- rweibull(a, b, c)
                ^~~~~~~~
================================================================================
[ FAIL 46 | WARN 22 | SKIP 0 | PASS 77 ]rxode2 model syntax error:
================================================================================
:001: 'riweibull'/'rxweibull'/'rweibull' takes 1-2 arguments 'rxweibull(shape, scale)':
          x1 <- rweibull()
                         ^
================================================================================
[ FAIL 86 | WARN 22 | SKIP 0 | PASS 174 ]

-- Warning (test-random.R:26:7): rnorm -----------------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x1), 0, tolerance = 0.01) test-random.R:26:6
 5. base::mean.default(f$x1)

-- Failure (test-random.R:26:7): rnorm -----------------------------------------
mean(f$x1) (`actual`) not equal to 0 (`expected`).

  `actual`: NA
`expected`:  0

-- Failure (test-random.R:27:7): rnorm -----------------------------------------
sd(f$x1) (`actual`) not equal to 1 (`expected`).

  `actual`: NA
`expected`:  1

-- Warning (test-random.R:29:7): rnorm -----------------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x2), 3, tolerance = 0.01) test-random.R:29:6
 5. base::mean.default(f$x2)

-- Failure (test-random.R:29:7): rnorm -----------------------------------------
mean(f$x2) (`actual`) not equal to 3 (`expected`).

  `actual`: NA
`expected`:  3

-- Failure (test-random.R:30:7): rnorm -----------------------------------------
sd(f$x1) (`actual`) not equal to 1 (`expected`).

  `actual`: NA
`expected`:  1

-- Warning (test-random.R:33:7): rnorm -----------------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x3), 5, tolerance = 0.01) test-random.R:33:6
 5. base::mean.default(f$x3)

-- Failure (test-random.R:33:7): rnorm -----------------------------------------
mean(f$x3) (`actual`) not equal to 5 (`expected`).

  `actual`: NA
`expected`:  5

-- Failure (test-random.R:34:7): rnorm -----------------------------------------
sd(f$x3) (`actual`) not equal to 2 (`expected`).

  `actual`: NA
`expected`:  2

-- Warning (test-random.R:121:7): rbinom ---------------------------------------
no non-missing arguments to max; returning -Inf
Backtrace:
 1. testthat::expect_equal(max(f$x1), 4) test-random.R:121:6
 2. testthat::quasi_label(enquo(object), label, arg = "object")
 3. rlang::eval_bare(expr, quo_get_env(quo))

-- Failure (test-random.R:121:7): rbinom ---------------------------------------
max(f$x1) (`actual`) not equal to 4 (`expected`).

  `actual`: -Inf
`expected`:    4

-- Warning (test-random.R:122:7): rbinom ---------------------------------------
no non-missing arguments to min; returning Inf
Backtrace:
 1. testthat::expect_equal(min(f$x1), 0) test-random.R:122:6
 2. testthat::quasi_label(enquo(object), label, arg = "object")
 3. rlang::eval_bare(expr, quo_get_env(quo))

-- Failure (test-random.R:122:7): rbinom ---------------------------------------
min(f$x1) (`actual`) not equal to 0 (`expected`).

  `actual`: Inf
`expected`:   0

-- Error (test-random.R:123:7): rbinom -----------------------------------------
Error in `round(f$x1)`: non-numeric argument to mathematical function
Backtrace:
 1. testthat::expect_true(all(round(f$x1) == f$x1)) test-random.R:123:6
 2. testthat::quasi_label(enquo(object), label, arg = "object")
 3. rlang::eval_bare(expr, quo_get_env(quo))

-- Warning (test-random.R:210:7): rchisq ---------------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x1), 15, tolerance = 0.1) test-random.R:210:6
 5. base::mean.default(f$x1)

-- Failure (test-random.R:210:7): rchisq ---------------------------------------
mean(f$x1) (`actual`) not equal to 15 (`expected`).

  `actual`: NA
`expected`: 15

-- Failure (test-random.R:211:7): rchisq ---------------------------------------
sd(f$x1) (`actual`) not equal to sqrt(2 * 15) (`expected`).

  `actual`: NA
`expected`:  5

-- Warning (test-random.R:213:7): rchisq ---------------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x2), 20, tolerance = 0.1) test-random.R:213:6
 5. base::mean.default(f$x2)

-- Failure (test-random.R:213:7): rchisq ---------------------------------------
mean(f$x2) (`actual`) not equal to 20 (`expected`).

  `actual`: NA
`expected`: 20

-- Failure (test-random.R:214:7): rchisq ---------------------------------------
sd(f$x2) (`actual`) not equal to sqrt(2 * 20) (`expected`).

  `actual`: NA
`expected`:  6

-- Warning (test-random.R:255:7): rexp tests -----------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x1), 2, tolerance = 0.1) test-random.R:255:6
 5. base::mean.default(f$x1)

-- Failure (test-random.R:255:7): rexp tests -----------------------------------
mean(f$x1) (`actual`) not equal to 2 (`expected`).

  `actual`: NA
`expected`:  2

-- Failure (test-random.R:256:7): rexp tests -----------------------------------
sd(f$x1) (`actual`) not equal to sqrt(1/(0.5 * 0.5)) (`expected`).

  `actual`: NA
`expected`:  2

-- Warning (test-random.R:258:7): rexp tests -----------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x2), 1, tolerance = 0.1) test-random.R:258:6
 5. base::mean.default(f$x2)

-- Failure (test-random.R:258:7): rexp tests -----------------------------------
mean(f$x2) (`actual`) not equal to 1 (`expected`).

  `actual`: NA
`expected`:  1

-- Failure (test-random.R:259:7): rexp tests -----------------------------------
sd(f$x2) (`actual`) not equal to 1 (`expected`).

  `actual`: NA
`expected`:  1

-- Warning (test-random.R:303:7): rf tests -------------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x1), mf(20), tolerance = 0.01) test-random.R:303:6
 5. base::mean.default(f$x1)

-- Failure (test-random.R:303:7): rf tests -------------------------------------
mean(f$x1) (`actual`) not equal to mf(20) (`expected`).

  `actual`: NA
`expected`:  1

-- Failure (test-random.R:304:7): rf tests -------------------------------------
sd(f$x1) (`actual`) not equal to sf(10, 20) (`expected`).

  `actual`: NA
`expected`:  1

-- Warning (test-random.R:306:7): rf tests -------------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x2), mf(40), tolerance = 0.01) test-random.R:306:6
 5. base::mean.default(f$x2)

-- Failure (test-random.R:306:7): rf tests -------------------------------------
mean(f$x2) (`actual`) not equal to mf(40) (`expected`).

  `actual`: NA
`expected`:  1

-- Failure (test-random.R:307:7): rf tests -------------------------------------
sd(f$x2) (`actual`) not equal to sf(30, 40) (`expected`).

  `actual`: NA
`expected`:  0

-- Failure (test-random.R:355:7): rgamma tests ---------------------------------
sd(f$x1) (`actual`) not equal to sgamma(9, 0.5) (`expected`).

  `actual`: NA
`expected`:  6

-- Failure (test-random.R:357:7): rgamma tests ---------------------------------
sd(f$x2) (`actual`) not equal to sgamma(7.5) (`expected`).

  `actual`: NA
`expected`:  3

-- Warning (test-random.R:405:7): rbeta tests ----------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x1), mbeta(2, 5), tolerance = 0.01) test-random.R:405:6
 5. base::mean.default(f$x1)

-- Failure (test-random.R:405:7): rbeta tests ----------------------------------
mean(f$x1) (`actual`) not equal to mbeta(2, 5) (`expected`).

  `actual`: NA
`expected`:  0

-- Failure (test-random.R:406:7): rbeta tests ----------------------------------
sd(f$x1) (`actual`) not equal to sbeta(2, 5) (`expected`).

  `actual`: NA
`expected`:  0

-- Warning (test-random.R:408:7): rbeta tests ----------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x2), mbeta(2, 2), tolerance = 0.01) test-random.R:408:6
 5. base::mean.default(f$x2)

-- Failure (test-random.R:408:7): rbeta tests ----------------------------------
mean(f$x2) (`actual`) not equal to mbeta(2, 2) (`expected`).

  `actual`: NA
`expected`:  0

-- Failure (test-random.R:409:7): rbeta tests ----------------------------------
sd(f$x2) (`actual`) not equal to sbeta(2, 2) (`expected`).

  `actual`: NA
`expected`:  0

-- Failure (test-random.R:454:7): rgeom tests ----------------------------------
median(f$x2) (`actual`) not equal to -ceiling(1/log2(1 - 0.1)) (`expected`).

`actual` is NULL
`expected` is a double vector (6)

-- Warning (test-random.R:491:7): rpois ----------------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x1), 1, tolerance = 0.01) test-random.R:491:6
 5. base::mean.default(f$x1)

-- Failure (test-random.R:491:7): rpois ----------------------------------------
mean(f$x1) (`actual`) not equal to 1 (`expected`).

  `actual`: NA
`expected`:  1

-- Failure (test-random.R:492:7): rpois ----------------------------------------
sd(f$x1) (`actual`) not equal to 1 (`expected`).

  `actual`: NA
`expected`:  1

-- Warning (test-random.R:494:7): rpois ----------------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x2), 2, tolerance = 0.01) test-random.R:494:6
 5. base::mean.default(f$x2)

-- Failure (test-random.R:494:7): rpois ----------------------------------------
mean(f$x2) (`actual`) not equal to 2 (`expected`).

  `actual`: NA
`expected`:  2

-- Failure (test-random.R:495:7): rpois ----------------------------------------
sd(f$x2) (`actual`) not equal to sqrt(2) (`expected`).

  `actual`: NA
`expected`:  1

-- Warning (test-random.R:534:7): rt -------------------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x1), 0, tolerance = 0.1) test-random.R:534:6
 5. base::mean.default(f$x1)

-- Failure (test-random.R:534:7): rt -------------------------------------------
mean(f$x1) (`actual`) not equal to 0 (`expected`).

  `actual`: NA
`expected`:  0

-- Failure (test-random.R:535:7): rt -------------------------------------------
sd(f$x1) (`actual`) not equal to sqrt(15/(15 - 2)) (`expected`).

  `actual`: NA
`expected`:  1

-- Warning (test-random.R:537:7): rt -------------------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x2), 0, tolerance = 0.1) test-random.R:537:6
 5. base::mean.default(f$x2)

-- Failure (test-random.R:537:7): rt -------------------------------------------
mean(f$x2) (`actual`) not equal to 0 (`expected`).

  `actual`: NA
`expected`:  0

-- Failure (test-random.R:538:7): rt -------------------------------------------
sd(f$x2) (`actual`) not equal to sqrt(20/(20 - 2)) (`expected`).

  `actual`: NA
`expected`:  1

-- Warning (test-random.R:582:7): runif ----------------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x1), 0.5, tolerance = 0.01) test-random.R:582:6
 5. base::mean.default(f$x1)

-- Failure (test-random.R:582:7): runif ----------------------------------------
mean(f$x1) (`actual`) not equal to 0.5 (`expected`).

  `actual`: NA
`expected`:  0

-- Failure (test-random.R:583:7): runif ----------------------------------------
sd(f$x1) (`actual`) not equal to sqrt(1/12) (`expected`).

  `actual`: NA
`expected`:  0

-- Warning (test-random.R:585:7): runif ----------------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x2), 0.5 * (0.5 + 1), tolerance = 0.01) test-random.R:585:6
 5. base::mean.default(f$x2)

-- Failure (test-random.R:585:7): runif ----------------------------------------
mean(f$x2) (`actual`) not equal to 0.5 * (0.5 + 1) (`expected`).

  `actual`: NA
`expected`:  1

-- Failure (test-random.R:586:7): runif ----------------------------------------
sd(f$x2) (`actual`) not equal to sqrt((1 - 0.5)^2/12) (`expected`).

  `actual`: NA
`expected`:  0

-- Warning (test-random.R:588:7): runif ----------------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x3), 0.5 * (0.25 + 0.75), tolerance = 0.01) test-random.R:588:6
 5. base::mean.default(f$x3)

-- Failure (test-random.R:588:7): runif ----------------------------------------
mean(f$x3) (`actual`) not equal to 0.5 * (0.25 + 0.75) (`expected`).

  `actual`: NA
`expected`:  0

-- Failure (test-random.R:589:7): runif ----------------------------------------
sd(f$x3) (`actual`) not equal to sqrt((0.75 - 0.25)^2/12) (`expected`).

  `actual`: NA
`expected`:  0

-- Warning (test-random.R:648:7): rweibull tests -------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x1), mweibull(9, 0.5), tolerance = 0.01) test-random.R:648:6
 5. base::mean.default(f$x1)

-- Failure (test-random.R:648:7): rweibull tests -------------------------------
mean(f$x1) (`actual`) not equal to mweibull(9, 0.5) (`expected`).

  `actual`: NA
`expected`:  0

-- Failure (test-random.R:649:7): rweibull tests -------------------------------
sd(f$x1) (`actual`) not equal to sweibull(9, 0.5) (`expected`).

  `actual`: NA
`expected`:  0

-- Warning (test-random.R:651:7): rweibull tests -------------------------------
argument is not numeric or logical: returning NA
Backtrace:
 1. testthat::expect_equal(mean(f$x2), mweibull(7.5), tolerance = 0.01) test-random.R:651:6
 5. base::mean.default(f$x2)

-- Failure (test-random.R:651:7): rweibull tests -------------------------------
mean(f$x2) (`actual`) not equal to mweibull(7.5) (`expected`).

  `actual`: NA
`expected`:  1

-- Failure (test-random.R:652:7): rweibull tests -------------------------------
sd(f$x2) (`actual`) not equal to sweibull(7.5) (`expected`).

  `actual`: NA
`expected`:  0

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:722:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

-- Failure (test-random.R:766:9): individual random variable tests -------------
sum(duplicated(paste0(f$id, f[[paste0("x", i)]]))) (`actual`) not equal to 5 (`expected`).

  `actual`: 0
`expected`: 5

[ FAIL 86 | WARN 22 | SKIP 0 | PASS 174 ]

Test complete
```

**Comment by mattfidler** [comment]
I am unsure what is happening here.  I will have to test on windows...

**Comment by mattfidler** [comment]
Strangely, it was succeeding with the other version of RxODE and the simulation code hasn't been altered.  Is there a possibility there is something strange about your compiler on windows? Maybe (accidently) using a different toolchain?

**Comment by mattfidler** [comment]
For some reason in these tests `expect-warning` returns the warning instead of the test results in windows.  I am working through this now

**Comment by mattfidler** [comment]
This is fixed for me.... Though I think it exposes a bug in `testthat`.  Check @billdenney  if it works for you when you have time

**Comment by mattfidler** [comment]
Also many of the "errors" in windows could be because warnings were expected.

**Comment by billdenney** [comment]
I've got a really big PR that is fixing a lot of these issues.  I'm going to make the PR in just a few minutes...

I think that you're right that expect_warning() returns a different value on different systems, and I'm about to work through testing that (and reporting, if correct).

**Comment by mattfidler** [comment]
It took me awhile to get to my windows system :)  I guess it will make merging this a bit more difficult.

**Comment by billdenney** [comment]
There was a bug in testthat 3.1.0, but it has been fixed in testthat 3.1.2.

**Comment by mattfidler** [comment]
Hm.  I installed a new R release with the new testthat today, and it still gave this bug.

**Comment by billdenney** [comment]
On R 4.1.2 on Windows 10 and Linux with testthat 3.1.2, both seemed to return the value of the function rather than the warning condition.  Given the discussion in r-lib/testthat#1371, I think that we should expect `expect_*()` functions not to return the value from the code inside the expression.

**Comment by mattfidler** [comment]
I think you are correct.  We cannot rely on the return values any more.

Note for me I did run with the newest testthat

**Comment by billdenney** [comment]
Alrighty, I've just made the biggest PR that I think I've ever made (because most of it is just line spacing changes).  #30 should fix this issue.

**Comment by mattfidler** [comment]
I think if you put "Closes #30" in the pull request comment, it will close the issue when you merge it.  I just found out recently.
